src/Forms/GridField/GridField.php

```diff
@@ -444,6 +451,63 @@ private function initState(): void
         }
     }
 
+    /**
+     * Adds state for this gridfield from the request variables.
+     *
+     * If there is state already set on this GridField, that takes precedent
+     * over state from the request.
+     */
+    private function addStateFromRequest(): void
+    {
+        $request = $this->getRequest();
+        if (($request instanceof NullHTTPRequest) && Controller::has_curr()) {
+            $request = Controller::curr()->getRequest();
+        }
+        // if ($request->params()['Action']) {
+        //     return;
+        // }
+        $stateStr = $this->getStateManager()->getStateFromRequest($this, $request);
+
+        if ($stateStr) {
+            $oldState = $this->getState(false);
+            // Create a dummy state so that we can merge the current state with the request state.
+            $newState = new GridState($this, $stateStr);
+            // Put the current state on top of the request state.
+            $newState->setValue($oldState->Value());
+            $this->state = $newState;
+        }
+    }
+
+    /**
+     * Add get and post parameters if exist to the URL.
+     * 
+     * URL keeps state for GridField, that exist in few different tabs.
+     */
+    public function addRequestToURL(string $link): string
+    {
+        $request = $this->getRequest();
+
+        $postparams = [];
+
+        if ($request->postVars()) {
+            foreach ($request->postVars() as $key => $val) {
+                if (stripos($key, 'gridState') === 0) {
+                    $postparams[$key] = $val;
+                }
+            }
+        }
+
+        $params = array_merge($request->getVars(), $postparams);
+
+        if ($params && is_array($params) && count($params) > 0) {
+            foreach ($params as $key => $val) {
+                $link = HTTP::setGetVar($key, $val, $link);
+            }  
+        }
+
+        return $link;
```

Comment:
```suggestion
        return $this->getStateManager()->addStateToURL($this, $link);
```
It seems like this works just as well, and doesn't require checking all of the vars? Is there some edge case that this doesn't catch?

Reply:
Actually given that this seems to be the same functionality, do we need it on `GridField` at all? Why not just keep calling `$this->getStateManager()->addStateToURL` directly?

Reply:
`addStateToURL` method just appends new state to URL, but it is not keeping previous requests. 
I can move this part in the GridFieldStateManager class, but I don't think this functionality is related to GridFieldStateManager entity.